### PR TITLE
Noloan admin app

### PR DESCRIPTION
**Pull request by gulilo**

### CODE REVIEW

File: projects/noloan/admin/app/src/main/java/hasadna/noloan/admin/app/DBMessagesHolder.java [hunk 1]
```diff
@@ -0,0 +1,103 @@
+package hasadna.noloan.admin.app;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import hasadna.noloan.protobuf.SmsProto.SmsMessage;
+
+// Simple singleton to hold the suggestions list
+public class DBMessagesHolder {
```
Comment (oferb) [inline projects/noloan/admin/app/src/main/java/hasadna/noloan/admin/app/DBMessagesHolder.java]: `DBMessagesHolder->DbMessagesHolder`

File: projects/noloan/admin/app/src/main/java/hasadna/noloan/admin/app/DBMessagesHolder.java [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+
+  public void addSpam(SmsMessage spam) {
+    this.spam.add(spam);
+
+    if (spamListener != null) {
+      spamListener.messageAdded();
+    }
+  }
+
+  public void spamModified(SmsMessage spam) {
```
Comment (oferb) [inline projects/noloan/admin/app/src/main/java/hasadna/noloan/admin/app/DBMessagesHolder.java]: If `spam` was changed, you won't be able to find it in `spam.indexOf()`.

File: projects/noloan/admin/app/src/main/java/hasadna/noloan/admin/app/SplashScreenActivity.java [hunk 3]
```diff
@@ -0,0 +1,121 @@
...
+        Tasks.await(Suggestions);
+      } catch (ExecutionException | InterruptedException e) {
+        e.printStackTrace();
+      }
+      if (!permissions.getResult()) {
+        handler.post(
+            () ->
+                Toast.makeText(
+                        getApplicationContext(),
+                        "This App require SMS reading To work",
```
Comment (oferb) [inline projects/noloan/admin/app/src/main/java/hasadna/noloan/admin/app/SplashScreenActivity.java]: Rephrase suggestion: `This App require SMS reading To work->This app requires permission to read SMSs`

File: projects/noloan/admin/app/src/main/java/hasadna/noloan/admin/app/firestore/FirestoreClient.java [hunk 4]
```diff
@@ -0,0 +1,118 @@
...
+
+  public void writeMessage(SmsMessage message, String path) {
+    FirestoreElement element = encodeMessage(message);
+    client.collection(path).add(element);
+  }
+
+  public void deleteMessage(SmsMessage sms, String path) {
+    client.collection(path).document(sms.getID()).delete().
+        addOnSuccessListener(aVoid -> {
+          Log.d("BLAAAAA","!!!!!!!!!!!!!!!!! success");
```
Comment (oferb) [inline projects/noloan/admin/app/src/main/java/hasadna/noloan/admin/app/firestore/FirestoreClient.java]: ? :-)

File: projects/noloan/admin/app/src/main/java/hasadna/noloan/admin/app/firestore/FirestoreClient.java [hunk 5]
```diff
@@ -0,0 +1,118 @@
...
+        })
+        .addOnFailureListener(e -> {
+          Log.d("BLAAAAA","!!!!!!!!!!!!!!!!! failed");
+        });
+  }
+
+  // Start real-time listening to the DB for change, return set the result to true when done.
+  public TaskCompletionSource StartListeningToMessages(String path) {
```
Comment (oferb) [inline projects/noloan/admin/app/src/main/java/hasadna/noloan/admin/app/firestore/FirestoreClient.java]: Methods should start with lowercase.

File: projects/noloan/admin/app/src/main/java/hasadna/noloan/admin/app/firestore/FirestoreClient.java [hunk 6]
```diff
@@ -0,0 +1,118 @@
...
+    CollectionReference collectionReference = client.collection(path);
+    collectionReference.addSnapshotListener(
+            executor,
+            (queryDocumentSnapshots, e) -> {
+              if (e != null) {
+                return;
+              }
+
+              DBMessagesHolder messagesHolder = DBMessagesHolder.getInstance();
+              for (DocumentChange dc : queryDocumentSnapshots.getDocumentChanges()) {
```
Comment (oferb) [inline projects/noloan/admin/app/src/main/java/hasadna/noloan/admin/app/firestore/FirestoreClient.java]: `dc->documentChange`

File: projects/noloan/admin/app/src/main/proto/COURT.proto [hunk 7]
```diff
@@ -0,0 +1,16 @@
+syntax = "proto3";
```
Comment (oferb) [inline projects/noloan/admin/app/src/main/proto/COURT.proto]: File name should be CamelCase, so `COURT->Court`. Same for `LAWSUIT` ans `SMS`.

File: projects/noloan/admin/app/src/main/proto/COURT.proto [hunk 8]
```diff
@@ -0,0 +1,16 @@
...
+option java_package = "hasadna.noloan.protobuf";
+option java_outer_classname = "CourtProto";
+
+message Court {
+    string name = 1;
+    string address = 2;
+    string fax= 3;
+    string website=4;
+}
+
```
Comment (oferb) [inline projects/noloan/admin/app/src/main/proto/COURT.proto]: nit: Redundant newlines. Also for `Lawsuit.proto`.

File: projects/noloan/admin/app/src/main/java/hasadna/noloan/admin/app/firestore/FirestoreClient.java (hunk 4 above)
Comment (gulilo) [reply projects/noloan/admin/app/src/main/java/hasadna/noloan/admin/app/firestore/FirestoreClient.java]: I said it need clean up

File: projects/noloan/admin/app/src/main/java/hasadna/noloan/admin/app/DBMessagesHolder.java (hunk 2 above)
Comment (gulilo) [reply projects/noloan/admin/app/src/main/java/hasadna/noloan/admin/app/DBMessagesHolder.java]: Tom fixed it in his PR

File: projects/noloan/admin/app/src/main/java/hasadna/noloan/admin/app/firestore/FirestoreClient.java (hunk 4 above)
Comment (gulilo) [reply projects/noloan/admin/app/src/main/java/hasadna/noloan/admin/app/firestore/FirestoreClient.java]: What do we want to happen on success and failed of deleted message from the DB?

File: projects/noloan/admin/app/src/main/java/hasadna/noloan/admin/app/SplashScreenActivity.java [hunk 9]
```diff
@@ -0,0 +1,121 @@
...
+        Tasks.await(Suggestions);
+      } catch (ExecutionException | InterruptedException e) {
+        e.printStackTrace();
+      }
+      if (!permissions.getResult()) {
+        handler.post(
+            () ->
+                Toast.makeText(
+                        getApplicationContext(),
+                        "This App require SMS reading To work->This app requires permission to read SMSs",
```
Comment (oferb) [inline projects/noloan/admin/app/src/main/java/hasadna/noloan/admin/app/SplashScreenActivity.java]: The `->` arrow means replace. So `A->B` means replace `A` with `B`.

File: projects/noloan/admin/app/src/main/java/hasadna/noloan/admin/app/DBMessagesHolder.java (hunk 2 above)
Comment (oferb) [reply projects/noloan/admin/app/src/main/java/hasadna/noloan/admin/app/DBMessagesHolder.java]: Ok

File: projects/noloan/admin/app/src/main/java/hasadna/noloan/admin/app/firestore/FirestoreClient.java (hunk 4 above)
Comment (oferb) [reply projects/noloan/admin/app/src/main/java/hasadna/noloan/admin/app/firestore/FirestoreClient.java]: On success - nothing.
On fail - log and show message to user.

File: projects/noloan/admin/app/src/main/proto/Sms.proto [hunk 10]
```diff
@@ -0,0 +1,17 @@
+syntax = "proto3";
```
Comment (oferb) [inline projects/noloan/admin/app/src/main/proto/Sms.proto]: Now I see that you copy pasted these protos.
I guess there's no clear way to use the ones from the main app?

**Pull request merged**